sdk/src/trace/recorder.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "opentelemetry/trace/span_id.h"
+#include "opentelemetry/trace/trace_id.h"
+#include "opentelemetry/version.h"
+#include "src/trace/recordable.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+class Recorder
+{
+public:
+  virtual ~Recorder() = default;
+
+  /**
+   * @return a Recordable object to maintain a data representation of a span.
+   */
+  virtual std::unique_ptr<Recordable> MakeRecordable() noexcept = 0;
+
+  /**
+   * Record a span.
+   * @param recordable the data representation of the span.
+   */
+  virtual void Record(std::unique_ptr<Recordable> &&recordable) noexcept = 0;
```

Comment:
Is there a reason for using `std::unique_ptr` instead of `nonstd::unique_ptr` in this interface?

Reply:
I was limiting nostd::unique_ptr to the places where we need a stable ABI that could cross DSOs. 

I don't know that are plans to support dynamically loadable Recorders. I feel like we should try to limit the points where we support plug-ability to avoid confusion and we already have two that we've discussed targeting (i.e. Tracer plugins and Exporter plugins)

Reply:
I think Recorder should be an implementation detail of the SDK.